torch/distributions/distribution.py

```diff
@@ -220,6 +241,19 @@ def _validate_sample(self, value):
         if not self.support.check(value).all():
             raise ValueError('The value argument must be within the support')
 
+    def _get_checked_instance(self, cls, instance=None):
+        def cons(cls_):
+            """
+            Returns the `__init__` function for a python 2.7 / 3 class.
+            """
+            return getattr(cls_.__init__, "__func__", cls_.__init__)
+
+        if instance is None and cons(type(self)) is not cons(cls):
+            raise NotImplementedError("Subclass {} of {} that defines a custom __init__ method "
+                                      "must also define a custom .expand() method.".
+                                      format(self.__class__.__name__, cls.__name__))
+        return self.__new__(type(self)) if not instance else instance
```

Comment:
It's safer to `... if instance is None else instance`, in case someone defines a `__bool__` method.